### PR TITLE
Acceptance tests: Add more fields in verify table calls used in other apps

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Acceptance tests: Add more fields in verify table calls used in other apps

## Related Issue

- Related https://github.com/owncloud/notifications/issues/300, https://github.com/owncloud/richdocuments/issues/312

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUINotificationsContext.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ public function assertNotificationsOnWebUI(
 			\count($notifications),
 			"expected $number notifications, found " . \count($notifications)
 		);
-		$this->featureContext->verifyTableNodeColumns($expectedNotifications, ['title'], ['content']);
+		$this->featureContext->verifyTableNodeColumns($expectedNotifications, ['title'], ['content', 'link', 'message']);
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUINotificationsContext.php]: how is `content` different to `message` ?

should we choose one of these and use it consistently?

File: tests/acceptance/features/bootstrap/WebUISharingContext.php [hunk 2]
```diff
@@ -756,7 +756,11 @@ public function theUserTypesInTheShareWithField($input) {
 	public function theUserSetsTheSharingPermissionsOfForOnTheWebUI(
 		$userOrGroup, $userName, $fileName, TableNode $permissionsTable
 	) {
-		$this->featureContext->verifyTableNodeRows($permissionsTable, [], ['share', 'edit', 'create', 'change', 'delete']);
+		$this->featureContext->verifyTableNodeRows(
+			$permissionsTable,
+			[],
+			['share', 'edit', 'create', 'change', 'delete', 'edit', 'secure-view', 'print']
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUISharingContext.php]: This is OK for now. But it does mean that core test code knows about the particular extra permission names in richdocuments. I  wonder how we can "pass in" the extra permission names that are allowed?

File: tests/acceptance/features/bootstrap/WebUINotificationsContext.php (hunk 1 above)
Comment (phil-davis) [reply tests/acceptance/features/bootstrap/WebUINotificationsContext.php]: @dpakach ping

File: tests/acceptance/features/bootstrap/WebUISharingContext.php (hunk 2 above)
Comment (skshetry) [reply tests/acceptance/features/bootstrap/WebUISharingContext.php]: I thought, the changes were regarding the missing headers case (and, that too where those headers are expected). This function does not really set the contract for what the headers should be, and therefore is not the concern of this function here. If certain arrays of data are required, those need to be verified by other functions (`setSharingPermissions` in this case). This is only making it more tightly coupled.

Comment (phil-davis) [reply tests/acceptance/features/bootstrap/WebUISharingContext.php]: for when there are rows of settings, we also made it possible to define what "names" (strings) are allowed in the left column of a row. This helps detect accidental typos in a table of share attributes to be set, or a table with a list of permission names.

Maybe an app that wants to use this kind of step can provide some method that informs about the extra valid values?

Comment (skshetry) [reply tests/acceptance/features/bootstrap/WebUISharingContext.php]: I see the valid use case of this in `postConditions`/`assertions`, but, not really on other places.
We do need to harden assertions functions, but, here I don't really see any gains (at least compared to the trouble).

Even here, if some attributes were **required**, I can see it's use, but this specific function does not.

**Review by phil-davis — APPROVED** [review]
Some CI is failing, and this is an OK first solution. Let's merge it to get CI green again in the apps.
I will make an issue to review this - #36684

**Pull request merged**